src/sql.rs

```diff
@@ -256,7 +253,7 @@ impl Sql {
         &self,
     ) -> Result<r2d2::PooledConnection<r2d2_sqlite::SqliteConnectionManager>> {
         let lock = self.pool.read().await;
-        let pool = lock.as_ref().ok_or(Error::SqlNoConnection)?;
+        let pool = lock.as_ref().ok_or(format_err!("No SQL connection"))?;
```

Comment:
Because this error is returned in so many branches I'm kind of tempted to leave it as an enum as it's easier on implementation and reading.  The public return type can stay `anyhow::Error` though (but it doesn't have to be).  And also I don't think the rest of the simplifications of this PR needs to change if you move this back to thiserror.

But overall I don't feel that strong on this, if you prefer this way.

Reply:
I have pushed more changes so now this error is returned only 2 times. Also not sure about `with_conn`, this could probably be removed.

Reply:
I rewrote `with_conn` using `get_conn`, now there is only one place which returns this error (`get_conn`).